src/SessionSetup/ConnectToTargetDialog.cpp

```diff
@@ -75,46 +66,66 @@ std::optional<TargetConfiguration> ConnectToTargetDialog::Exec() {
     return std::nullopt;
   }
 
-  return target;
+  return std::move(target_configuration_);
 }
 
-ErrorMessageOr<StadiaTarget> ConnectToTargetDialog::OnAsyncDataAvailable(
-    MaybeSshAndInstanceData ssh_instance_data) {
-  OUTCOME_TRY(auto&& ssh_info, std::get<0>(ssh_instance_data));
-  OUTCOME_TRY(auto&& instance, std::get<1>(ssh_instance_data));
+void ConnectToTargetDialog::OnAsyncDataAvailable(MaybeSshAndInstanceData ssh_instance_data) {
+  auto maybe_ssh_info = std::get<0>(ssh_instance_data);
+
+  if (maybe_ssh_info.has_error()) {
+    LogAndDisplayError(maybe_ssh_info.error());
+    reject();
+    return;
+  }
+  orbit_ggp::SshInfo ssh_info = std::move(maybe_ssh_info.value());
+
+  auto maybe_instance_data = std::get<1>(ssh_instance_data);
 
-  ConnectionData connection_data;
+  if (maybe_instance_data.has_error()) {
+    LogAndDisplayError(maybe_instance_data.error());
+    reject();
+    return;
+  }
+  orbit_ggp::Instance instance = std::move(maybe_instance_data.value());
 
-  connection_data.service_deploy_manager_ =
-      std::make_unique<orbit_session_setup::ServiceDeployManager>(
-          ssh_connection_artifacts_->GetDeploymentConfiguration(),
-          ssh_connection_artifacts_->GetSshContext(), CredentialsFromSshInfo(ssh_info),
-          ssh_connection_artifacts_->GetGrpcPort());
+  auto service_deploy_manager = std::make_unique<orbit_session_setup::ServiceDeployManager>(
+      ssh_connection_artifacts_->GetDeploymentConfiguration(),
+      ssh_connection_artifacts_->GetSshContext(), CredentialsFromSshInfo(ssh_info),
+      ssh_connection_artifacts_->GetGrpcPort());
 
-  OUTCOME_TRY(auto&& grpc_port, DeployOrbitService(connection_data.service_deploy_manager_.get()));
-  connection_data.grpc_channel_ = CreateGrpcChannel(grpc_port.grpc_port);
+  auto maybe_grpc_port = DeployOrbitService(service_deploy_manager.get());
 
-  OUTCOME_TRY(connection_data.process_data_,
-              FindSpecifiedProcess(connection_data.grpc_channel_, target_.process_name_or_path));
+  if (maybe_grpc_port.has_error()) {
+    LogAndDisplayError(maybe_grpc_port.error());
+    reject();
+    return;
+  }
 
-  return CreateTarget(std::move(connection_data), std::move(instance));
+  auto grpc_channel = CreateGrpcChannel(maybe_grpc_port.value().grpc_port);
+  stadia_connection_ = orbit_session_setup::StadiaConnection(
+      std::move(instance), std::move(service_deploy_manager), std::move(grpc_channel));
+
+  process_manager_ = orbit_client_services::ProcessManager::Create(
+      stadia_connection_.value().GetGrpcChannel(), absl::Milliseconds(1000));
+  process_manager_->SetProcessListUpdateListener(
+      [this](std::vector<orbit_grpc_protos::ProcessInfo> process_list) {
+        OnProcessListUpdate(std::move(process_list));
+      });
+  SetStatusMessage("Waiting for process to launch.");
```

Comment:
pers pref: Display `target_.process_name_or_path` in this message

Reply:
There's a label right above this message that displays the process name, so I think this is not needed here.

Reply:
Did not realize this. Then please ignore